Apps/phdnsdb/dnsdb_connector.py

```diff
@@ -343,91 +344,360 @@ def _lookup_ip(self, param):
+                                                        time_first_after=timestamps[1],
+                                                        time_last_before=timestamps[2],
+                                                        time_last_after=timestamps[3],
+                                                        ignore_limited=True))
+        except dnsdb2.exceptions.AccessDenied:
+            return action_result.set_status(
+                phantom.APP_ERROR, DNSDB_REST_RESP_ACCESS_DENIED_MSG)
+        except dnsdb2.exceptions.QuotaExceeded:
+            return action_result.set_status(
+                phantom.APP_ERROR, DNSDB_REST_RESP_LIC_EXCEED_MSG)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            ret_val = action_result.set_status(phantom.APP_ERROR, err)
+
+        # Something went wrong with the request
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # No data is considered as app success
+        if len(responses) == 0:
+            return action_result.set_status(phantom.APP_SUCCESS, DNSDB_DATA_NOT_AVAILABLE_MSG)
+
+        # To display count of domains in summary data
+        count_domain = set()
+
+        for resp in responses:
+
+            if DNSDB_JSON_RRNAME in resp:
+                resp[DNSDB_JSON_RRNAME] = resp[DNSDB_JSON_RRNAME].rstrip('.')
+                count_domain.add(resp[DNSDB_JSON_RRNAME])
+
+            # Response from the API is list of rdata.
+            # Adding Each data of list to action_result
+            action_result.add_data(resp)
+
+        summary_data['total_domains'] = len(count_domain)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _flex_search(self, param):
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Getting mandatory input parameter
+        query = param[DNSDB_JSON_QUERY]
+        rrtype = param[DNSDB_JSON_TYPE]
+        if rrtype not in DNSDB_JSON_TYPE_VALUE_LIST:
+            return action_result.set_status(phantom.APP_ERROR, DNSDB_VALUE_LIST_VALIDATION_MSG.format(DNSDB_JSON_TYPE_VALUE_LIST, DNSDB_JSON_TYPE))
+        search_type = param[DNSDB_JSON_SEARCH_TYPE]
+        if search_type not in DNSDB_JSON_SEARCH_TYPE_VALUE_LIST:
+            return action_result.set_status(phantom.APP_ERROR, DNSDB_VALUE_LIST_VALIDATION_MSG.format(DNSDB_JSON_SEARCH_TYPE_VALUE_LIST, DNSDB_JSON_SEARCH_TYPE))
+
+        # Getting optional input parameter
+        limit = param.get(DNSDB_JSON_LIMIT, 10000)
+        time_first_after = param.get(DNSDB_JSON_TIME_FIRST_AFTER)
+        time_first_before = param.get(DNSDB_JSON_TIME_FIRST_BEFORE)
+        time_last_after = param.get(DNSDB_JSON_TIME_LAST_AFTER)
+        time_last_before = param.get(DNSDB_JSON_TIME_LAST_BEFORE)
+        timestamps = [time_first_before, time_first_after, time_last_before, time_last_after]
+        for i, timestamp in enumerate(timestamps):
```

Comment:
Please keep this time fencing parameters validation in one common method _validate_params to avoid repetition of code lines. Pls, apply this for all the occurrences.

Reply:
Added this in _validate_params method